pkg/execution/state/redis_state/lua/queue/backlogPrepareNormalize.lua

```diff
@@ -9,29 +9,53 @@
 
   1 - Moved backlog to normalize set
   -1 - Fewer items than minimum
+  -2 - Garbage-collected empty backlog
 ]]
 
-local keyBacklogSet                      = KEYS[1]
-local keyShadowPartitionSet              = KEYS[2]
-local keyGlobalShadowPartitionSet        = KEYS[3]
-local keyGlobalAccountShadowPartitionSet = KEYS[4]
-local keyAccountShadowPartitionSet       = KEYS[5]
+local keyBacklogMeta                     = KEYS[1]
+local keyShadowPartitionMeta             = KEYS[2]
 
-local keyGlobalNormalizeSet              = KEYS[6]
-local keyAccountNormalizeSet             = KEYS[7]
-local keyPartitionNormalizeSet           = KEYS[8]
+local keyBacklogSet                      = KEYS[3]
+local keyShadowPartitionSet              = KEYS[4]
+local keyGlobalShadowPartitionSet        = KEYS[5]
+local keyGlobalAccountShadowPartitionSet = KEYS[6]
+local keyAccountShadowPartitionSet       = KEYS[7]
+
+local keyGlobalNormalizeSet              = KEYS[8]
+local keyAccountNormalizeSet             = KEYS[9]
+local keyPartitionNormalizeSet           = KEYS[10]
 
 local backlogID             = ARGV[1]
 local partitionID           = ARGV[2]
 local accountID             = ARGV[3]
 local normalizeTime         = tonumber(ARGV[4])
 local normalizeAsyncMinimum = tonumber(ARGV[5])
 
+-- $include(update_pointer_score.lua)
+-- $include(ends_with.lua)
+-- $include(update_account_queues.lua)
+-- $include(update_backlog_pointer.lua)
+
+local backlogCount = redis.call("ZCARD", keyBacklogSet)
+
+-- If backlog is empty, garbage-collect it from shadow partition
+if backlogCount == nil or backlogCount == false or backlogCount == 0 then
+  -- Remove pointer from shadow partition
+  redis.call("ZREM", keyShadowPartitionSet, backlogID)
+
+  -- Remove meta
+  redis.call("HDEL", keyBacklogMeta, backlogID)
```

Comment:
looks like these lines were called before the `updateBacklogPointer` fn, so the cleanup already happened here.